### ISSUE TITLE
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.

### ISSUE THREAD

**Issue by aaryadevchandra** [PowerShell/PSReadLine#2261]
![image](https://user-images.githubusercontent.com/66318555/109432384-e55c9180-7a30-11eb-8932-87b0f37508c8.png)

**Comment by abrighton-mc** [comment]
Same issue happening for me, looks like related to the contents of the history file @ `(Get-PSReadlineOption).HistorySavePath`.

I resolved the issue by manually removing a bunch of multi-line "Invoke-WebRequest -Uri" requests from the above file.

To reproduce:

1. Run the following (for example) Invoke-WebRequest command:

```
Invoke-WebRequest -Uri "https://github.com/manifest.json" -Headers @{
"method"="GET"
  "authority"="github.com"
  "scheme"="https"
  "path"="/manifest.json"
  "sec-ch-ua"="`"Chromium`";v=`"88`", `"Google Chrome`";v=`"88`", `";Not A Brand`";v=`"99`""
  "sec-ch-ua-mobile"="?0"
  "user-agent"="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36"
  "accept"="*/*"
  "sec-fetch-site"="same-origin"
  "sec-fetch-mode"="cors"
  "sec-fetch-dest"="empty"
  "referer"="https://github.com/PowerShell/PSReadLine/issues/2261"
  "accept-encoding"="gzip, deflate, br"
  "accept-language"="en-US,en;q=0.9"
  "cookie"="***OMITTED***"
}
```

2. Press Ctrl+R and type something (such as npm)
3. Bug occurs.

It's a fairly annoying bug actually, because you can't seem to escape the Ctrl+R reverse look up once the error occurs, no amount of backspacing, or pressing Ctrl+C has any effect.  In one instance, it actually locked up my VSCode instance and I had to task manager kill it.

**Comment by aaronbrighton** [comment]
Same issue happening for me, looks like related to the contents of the history file @ (Get-PSReadlineOption).HistorySavePath.

I resolved the issue by manually removing a bunch of multi-line "Invoke-WebRequest -Uri" requests from the above file.

To reproduce:

1. Run the following (for example) Invoke-WebRequest command:

```
Invoke-WebRequest -Uri "https://github.com/manifest.json" -Headers @{
"method"="GET"
  "authority"="github.com"
  "scheme"="https"
  "path"="/manifest.json"
  "sec-ch-ua"="`"Chromium`";v=`"88`", `"Google Chrome`";v=`"88`", `";Not A Brand`";v=`"99`""
  "sec-ch-ua-mobile"="?0"
  "user-agent"="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36"
  "accept"="*/*"
  "sec-fetch-site"="same-origin"
  "sec-fetch-mode"="cors"
  "sec-fetch-dest"="empty"
  "referer"="https://github.com/PowerShell/PSReadLine/issues/2261"
  "accept-encoding"="gzip, deflate, br"
  "accept-language"="en-US,en;q=0.9"
  "cookie"="***OMITTED***"
}
```

2. Press Ctrl+R and type something (such as npm)
3. Bug occurs.

It's a fairly annoying bug actually, because you can't seem to escape the Ctrl+R reverse look up once the error occurs, no amount of backspacing, or pressing Ctrl+C has any effect. In one instance, it actually locked up my VSCode instance and I had to task manager kill it.

**Comment by aaryadevchandra** [comment]
Ahh I see, so that's what's happening...thanks!

**Comment by daxian-dbw** [comment]
This issue was already fixed (see #1306). Please upgrade to the [2.1.0 version of PSReadLine](https://www.powershellgallery.com/packages/PSReadLine/2.1.0) from PowerShell Gallery.
See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions. Please let us know if you run into the same issue with the latest version.

**Comment by aaronbrighton** [comment]
> This issue was already fixed (see #1306). Please upgrade to the [2.1.0 version of PSReadLine](https://www.powershellgallery.com/packages/PSReadLine/2.1.0) from PowerShell Gallery.
> See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions. Please let us know if you run into the same issue with the latest version.

For anyone who doesn't remember ever explicitly installing PSReadLine, seems I was able to resolve this by simply going to the Help menu in VSCode and "Check for updates...".

Thanks!
